blaze/compute/sql.py

```diff
@@ -176,6 +176,12 @@ def compute_up(t, s, **kwargs):
     return getattr(t, 'op', getattr(safuncs, sym, getattr(sa.func, sym)))(s)
 
 
+@dispatch(Selection, sa.sql.ColumnElement)
+def compute_up(expr, data, scope=None, **kwargs):
+    predicate = compute(expr.predicate, data, post_compute=False)
+    return sa.select([data]).where(predicate)
```

Comment:
Sounds good. I just wanted to make sure that this wasn't solving some subtle issue that has been discovered in the past. I would also be in favor of removing at it is confusing when used with `sa.select`